### PR TITLE
🧪 Add tests for ParseArchesDescFile

### PR DESCRIPTION
**Pull request by arran4**
🎯 **What:** The testing gap addressed: Added tests for the previously untested public function `ParseArchesDescFile` in `arches.go`.
📊 **Coverage:** What scenarios are now tested:
- Happy path (file exists and is parsed correctly)
- File not found (`os.IsNotExist` fallback behavior correctly returning default headers)
- Standard error (simulated by passing a directory path as a file)
✨ **Result:** The improvement in test coverage: Code reliability is increased and edge case behavior is verified.

---
*PR created automatically by Jules for task [14698281395285688077](https://jules.google.com/task/14698281395285688077) started by @arran4*

### CODE REVIEW

File: arches_test.go [hunk 1]
```diff
@@ -141,3 +143,60 @@ x86 stable
...
+func TestParseArchesDescFile(t *testing.T) {
+	tempDir := t.TempDir()
+
+	// Test Case 1: File exists
+	validContent := `# Header
+amd64 stable
+x86 testing
+`
+	validFile := filepath.Join(tempDir, "arches.desc")
+	err := os.WriteFile(validFile, []byte(validContent), 0644)
```
Comment (arran4) [inline arches_test.go]: @jules do not perform real file operations and test refactor so we can inject a mock file system for test purposes
